src/psyclone/tests/psyir/transformations/reference2arrayrange_trans_test.py

```diff
@@ -346,6 +346,23 @@ def test_validate_deallocate(fortran_reader):
             " be transformed, but found:\n DEALLOCATE(a)" in str(info.value))
 
 
+def test_validate_pointer_assignment(fortran_reader):
+    '''Test that a reference in a PointerAssignment raises an exception. '''
+    code = (
+        "program test\n"
+        "  integer, dimension(10), target :: a\n"
+        "  integer, dimension(10), pointer :: b\n"
+        "  b => a\n"
```

Comment:
Just to be on the safe side, please also test with e.g. "this%that => a" and "b => this%that"

Reply:
StructureReferences are not accepted and will fail before reaching this point, I added the requested checks in test_validate_structure instead.